MTM-Frontend/src/pages/Auth/Register.tsx

```diff
@@ -5,30 +5,61 @@ import { Link, useNavigate } from "react-router-dom";
 import * as Yup from "yup";
 import { useAuth } from "../../AuthContext";
 import FormError from "./FormError";
-import { TextField, Button, Typography, Box } from "@mui/material";
+import {
+  Button,
+  Typography,
+  Box,
+  FormHelperText,
+  FormControl,
+  Select,
+  MenuItem,
+} from "@mui/material";
 
+// Register components
+import { RegisterTextField } from "../../components/RegisterForms/RegisterTextField";
+import { RegisterTextFieldPassword } from "../../components/RegisterForms/RegisterTextFieldPassword";
 interface FormValues {
-  firstName: string;
-  lastName: string;
+  name: string;
   email: string;
   password: string;
   confirmPassword: string;
   userType: string;
+
+  phone: string;
+  address: string;
+  zip: string;
+  city: string;
+  agency?: string;
 }
 
 const schema = Yup.object().shape({
-  firstName: Yup.string().required("First Name is required"),
-  lastName: Yup.string().required("Last Name is required"),
+  name: Yup.string()
+    .matches(/^([A-Za-z]+\s[A-Za-z]+)$/, "First and Last name")
+    .required("First and Last name is required"),
   email: Yup.string()
     .email("Invalid email address")
     .required("Email is required"),
   password: Yup.string()
+    .matches(
+      /^(?=.*[0-9])(?=.*[!@#$%^&*])[A-Za-z0-9!@#$%^&*]{8,}$/,
+      "Invalid password",
```

Comment:
If you are adding password rules make sure it gives the user appropriate error message when they input invalid password (eg. must be 9 chars long)

Reply:
In the password input, underneath it there's a message for 1 special and 1 number character.